src/Catalyst.Core.Modules.Dfs.Tests/Utils/TestDfs.cs

```diff
@@ -0,0 +1,112 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using Autofac;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.Dfs.Migration;
+using Catalyst.Abstractions.FileSystem;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Core.Lib.Cryptography;
+using Catalyst.Core.Modules.Dfs.Migration;
+using Catalyst.Core.Modules.Hashing;
+using Catalyst.Core.Modules.Keystore;
+using Catalyst.TestUtils;
+using Newtonsoft.Json.Linq;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.Utils
+{
+    public class TestDfs
+    {
+        private sealed class TestDfsFileSystem : FileSystemBasedTest
+        {
+            internal TestDfsFileSystem(ITestOutputHelper output) : base(output) { }
+        }
+
+        public static IDfsService GetTestDfs(ITestOutputHelper output, IFileSystem fileSystem = default, string keyType = default, IHashProvider hashProvider = null)
+        {
+            var type = output.GetType();
+            var testMember = type.GetField("test", BindingFlags.Instance | BindingFlags.NonPublic);
+            var test = (ITest) testMember.GetValue(output);
+
+            output.WriteLine("*************************** CI TEST:" + test.DisplayName + "******************************");
+            var nodeGuid = Guid.NewGuid();
+            var containerBuilder = new ContainerBuilder();
+
+            //if (string.Equals(folderName, default, StringComparison.Ordinal))
+            //{
+            //    folderName = nodeGuid.ToString();
+            //}
+
+            if (keyType == default)
+            {
+                keyType = "rsa";
+
+                //"ed25519";
+            }
+
+            if (fileSystem == null)
+            {
+                fileSystem = new TestDfsFileSystem(output).FileSystem;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'fileSystem'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)